src/onegov/ticket/handler.py

```diff
@@ -212,6 +212,26 @@ def get_links(self, request: 'CoreRequest') -> 'Sequence[_LinkOrCallback]':
 
         raise NotImplementedError
 
+    @property
+    def ticket_deletable(self) -> bool:
+        if self.deleted:
+            return True
+        if self.ticket.state != 'archived':
+            return False
+        if self.payment:
+            # For now we do not handle this case since payment might be
+            # needed for exports
+            return False
+        if self.undecided:
+            return False
+        return True
+
+    def prepare_delete_ticket(self) -> None:
+        """The handler knows best what to do when a ticket is called for
+        deletion. """
+        assert self.ticket_deletable
```

Comment:
This assertion suggest to me that we might want `ticket_deletable` to change to a `cached_property` in some cases, otherwise we do the work of figuring out if the ticket is deletable twice.

Reply:
I'm kind of questioning if the assertion is even worth it to write. That was there before in  `prepare_delete_ticket`, I did not add this

Reply:
It's a pretty good sanity check to make sure this only gets called for tickets that are actually deletable.